crates/bevy_ecs/src/event/registry.rs

```diff
@@ -77,4 +103,25 @@ impl EventRegistry {
             }
         }
     }
+
+    /// Removes an event from the world and it's associated EventRegistry
+    pub fn deregister_events<T: Event>(world: &mut World) {
+        let component_id = world.init_resource::<Events<T>>();
+        let mut registry = world.get_resource_or_insert_with(Self::default);
+        registry
+            .event_updates
+            .retain(|e| e.component_id != component_id);
+        world.remove_resource::<Events<T>>();
+    }
+
+    /// Removes an event from this registry and world
+    ///
+    /// In contrast, [`EventRegistry::deregister_events<T>`] will remove the events from the world's
+    /// main event registry
+    pub fn deregister_events_with_instance<T: Event>(&mut self, world: &mut World) {
```

Comment:
The with_instance naming isn't clear to me: is there an alternative way to frame this?

Reply:
The ideal way to frame it, in my opinion, would be using something like:

- renaming the  `register_events` & `deregister_events` to `register_events_in_default_registry`
- renaming the `with_instance` variants to the current names of the main ones

But since it involves essentially boils down to changing the behaviour of the currently named items, that will be a migration pain point - and not something I'd like to do right before a release.

I've also realized that there is much more work necessary to allow some way of actually using more than 1 EventRegistry in a world. So I think I'll take those elements out and return this to the simpler solution it to begin with.

Reply:
I have a concept I've been mulling over, and will likely set up some kind of discussion around it in the near future that will include this.